include/swift/AST/DiagnosticsDriver.def

```diff
@@ -156,8 +156,8 @@ WARNING(warn_opt_remark_disabled, none,
 WARNING(warn_ignoring_batch_mode,none,
 "ignoring '-enable-batch-mode' because '%0' was also specified", (StringRef))
 
-WARNING(warn_ignoring_single_compile, none,
-        "ignoring '%0' because '-dump-ast' was also specified", (StringRef))
+WARNING(warn_ignoring_wmo, none,
+        "ignoring '-wmo' because '-dump-ast' was also specified", ())
```

Comment:
I think this was better the way it was before, since there are several ways to spell the flag.

Reply:
The place from which I copied this code (Driver.cpp at line 1672, just below my changes) used `ArgRequiringSingleCompile->getOption().getPrefixedName()`. As far as I can tell, this will always print "`-whole-module-optimization`", even if you pass the shortened `-wmo` version.

The snippet I copied uses this method to distinguish between printing `-whole-module-optimization` or `-index-file`, but I don't think it'll have the desired effect here unless we change the way we get the flag name too.

Reply:
Hm, then we should probably be using `ArgRequiringSingleCompile->getSpelling()` instead, probably in both places.